tests/src/Functional/IvwIntegrationOverrideTest.php

```diff
@@ -0,0 +1,205 @@
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create and log in an administrative user.
+    $this->adminUser = $this->drupalCreateUser([
+      'create ivw_test content',
+      'administer ivw integration configuration',
+      'create terms in ivw_taxonomy',
+    ]);
+
+    // Setting defaults. To provide sane defaults and have a quick reference for
+    // the actual tests.
+    $this->config('ivw_integration.settings')->set('site', 'TestSiteName')
+      ->set('mobile_site', 'TestMobileSiteName')
+      ->set('frabo_default', 'IN')
+      ->set('frabo_mobile_default', 'mo')
+      ->set('frabo_overridable', 0)
+      ->set('frabo_mobile_overridable', 0)
+      ->set('code_template', '[ivw:offering]L[ivw:language]F[ivw:format]S[ivw:creator]H[ivw:homepage]D[ivw:delivery]A[ivw:app]P[ivw:paid]C[ivw:content]')
+      ->set('responsive', 1)
+      ->set('mobile_width', 480)
+      ->set('offering_default', '01')
+      ->set('offering_overridable', 0)
+      ->set('language_default', 1)
+      ->set('language_overridable', 0)
+      ->set('format_default', 1)
+      ->set('format_overridable', 0)
+      ->set('creator_default', 1)
+      ->set('creator_overridable', 0)
+      ->set('homepage_default', 2)
+      ->set('homepage_overridable', 0)
+      ->set('delivery_default', 1)
+      ->set('delivery_overridable', 0)
+      ->set('app_default', 1)
+      ->set('app_overridable', 0)
+      ->set('paid_default', 1)
+      ->set('paid_overridable', 0)
+      ->set('content_default', '01')
+      ->set('content_overridable', 0)
+      ->set('mcvd', 0)
+      ->save();
+
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests overriding of site values.
+   *
+   * @dataProvider overrideTestCases
+   */
+  public function testOverride($settings, $termOverrides, $nodeOverrides, $expectedOutput) {
+
+    if (!empty($settings)) {
+      $ivwSettings = $this->config('ivw_integration.settings');
+      foreach ($settings as $settingName => $settingValue) {
+        $ivwSettings->set($settingName, $settingValue);
+      }
+      $ivwSettings->save();
+    }
+
+    if (!empty($termOverrides)) {
+      // Load the term edit page.
+      $this->drupalGet('admin/structure/taxonomy/manage/ivw_taxonomy/add');
+      $this->assertSession()->statusCodeEquals(200);
+
+      $edit = [
+        'name[0][value]' => $this->randomString(),
+      ];
+
+      foreach ($termOverrides as $termOverrideName => $termOverrideValue) {
+        $edit["field_ivw_settings[0][$termOverrideName]"] = $termOverrideValue;
+      }
+      $this->drupalPostForm(NULL, $edit, 'Save');
+    }
+
+    // Load the node edit page.
+    $this->drupalGet('node/add/ivw_test');
+    $this->assertSession()->statusCodeEquals(200);
+
+    $edit = [
+      'title[0][value]' => $this->randomString(),
+    ];
+
+    if (!empty($termOverrides)) {
+      $edit['field_a'] = 1;
```

Comment:
Instead of assuming that this is ID 1, better load the term by `$terms = taxonomy_term_load_multiple_by_name($term->getName());`